src/test/java/it/gov/pagopa/reward/notification/service/rewards/evaluate/notify/RewardsNotificationExpiredInitiativeHandlerServiceIntegrationTest.java

```diff
@@ -0,0 +1,161 @@
+                RewardsNotificationFaker.mockInstanceBuilder(2)
+                        .initiativeId("INITIATIVE1")
+                        .notificationDate(null)
+                        .build()
+        ).block());
+        notificationsTestData.add(notificationRepository.save(
+                RewardsNotificationFaker.mockInstanceBuilder(3)
+                        .initiativeId("INITIATIVE2")
+                        .notificationDate(TODAY)
+                        .build()
+        ).block());
+        notificationsTestData.add(notificationRepository.save(
+                RewardsNotificationFaker.mockInstanceBuilder(4)
+                        .initiativeId("INITIATIVE2")
+                        .notificationDate(null)
+                        .build()
+        ).block());
+        notificationsTestData.add(notificationRepository.save(
+                RewardsNotificationFaker.mockInstanceBuilder(5)
+                        .initiativeId("INITIATIVE4")
+                        .notificationDate(TODAY)
+                        .build()
+        ).block());
+        notificationsTestData.add(notificationRepository.save(
+                RewardsNotificationFaker.mockInstanceBuilder(6)
+                        .initiativeId("INITIATIVE4")
+                        .notificationDate(null)
+                        .build()
+        ).block());
+
+    }
+
+    @BeforeEach
+    void init() {
+        service = new RewardsNotificationExpiredInitiativeHandlerServiceImpl(ruleRepository, notificationRepository, dayBefore);
+        storeTestData();
+    }
+
+    @AfterEach
+    void clearData() {
+        ruleRepository.deleteAll(initiativesTestData).block();
+        notificationRepository.deleteAll(notificationsTestData).block();
+    }
+
+    @Test
+    void testHandle() {
+        // Given
+        RewardsNotification expectedNotification = notificationsTestData.get(1);
+
+        // When
+        List<RewardsNotification> result = service.handle().collectList().block();
+
+        // Then
+        Assertions.assertNotNull(result);
+        Assertions.assertEquals(1, result.size());
+        checkHandleResult(expectedNotification, result.get(0));
+        checkResultFromRepo(result);
+    }
+
+    void checkResultFromRepo(List<RewardsNotification> result) {
+        List<String> resultIds = new ArrayList<>();
+
+        for(RewardsNotification n : result) {
+            resultIds.add(n.getId());
+            RewardsNotification entry = notificationRepository.findById(n.getId()).block();
+            Assertions.assertNotNull(entry);
+            Assertions.assertEquals(n, entry);
+        }
+
+        for (RewardsNotification n : notificationsTestData) {
+            if (!resultIds.contains(n.getId())) {
+                RewardsNotification entry = notificationRepository.findById(n.getId()).block();
+                Assertions.assertNotNull(entry);
+                Assertions.assertEquals(n, entry);
+            }
+        }
```

Comment:
you have also to check that all the other test data (thus not contained in the results) are not been modified (thus they are equal to notificationsTestData items)